### PR TITLE
AG-8260 Radial Bar series

### PR DESCRIPTION
**Pull request by alexanderby**

### CODE REVIEW

File: packages/ag-charts-enterprise/src/axes/angle-number/angleNumberAxis.ts [hunk 1]
```diff
@@ -0,0 +1,468 @@
...
+    (v: any) => ANGLE_LABEL_ORIENTATIONS.includes(v),
+    `expecting a label orientation keyword such as 'fixed', 'parallel' or 'perpendicular'`
+);
+
+class AngleCategoryAxisLabel extends _ModuleSupport.AxisLabel {
+    @Validate(ANGLE_LABEL_ORIENTATION)
+    orientation: AgAngleAxisLabelOrientation = 'fixed';
+}
+
+export class AngleNumberAxis extends _ModuleSupport.PolarAxis<_Scale.LinearScale> {
```
Comment (alantreadway) [inline packages/ag-charts-enterprise/src/axes/angle-number/angleNumberAxis.ts]: This entire class looks very similar to `angleCategoryAxis.ts` - can we factor out the common behaviours?

File: packages/ag-charts-enterprise/src/axes/radius-category/radiusCategoryAxis.ts [hunk 2]
```diff
@@ -0,0 +1,200 @@
...
+}
+
+type TickDatum = {
+    tickLabel: string;
+    tick: any;
+    tickId: string;
+    translationY: number;
+};
+
+export class RadiusCategoryAxis extends _ModuleSupport.PolarAxis {
```
Comment (alantreadway) [inline packages/ag-charts-enterprise/src/axes/radius-category/radiusCategoryAxis.ts]: There seems to be a bunch of commonality with `radiusNumberAxis.ts` - can we also factor out some common code?

File: packages/ag-charts-enterprise/src/axes/radius-category/radiusCrossLine.ts [hunk 3]
```diff
@@ -0,0 +1,157 @@
...
+const { ChartAxisDirection, Validate, OPT_NUMBER } = _ModuleSupport;
+const { Path, Sector, Text } = _Scene;
+const { normalizeAngle360, toRadians, isNumberEqual } = _Util;
+
+class RadiusCrossLineLabel extends PolarCrossLineLabel {
+    @Validate(OPT_NUMBER(-360, 360))
+    positionAngle?: number = undefined;
+}
+
+export class RadiusCrossLine extends PolarCrossLine {
```
Comment (alantreadway) [inline packages/ag-charts-enterprise/src/axes/radius-category/radiusCrossLine.ts]: This seems to be an identical implementation to the `radius-number` axis implementation? Can we factor out the common code please?

File: packages/ag-charts-enterprise/src/axes/angle-number/angleCrossLine.ts [hunk 4]
```diff
@@ -0,0 +1,173 @@
+import { _ModuleSupport, _Scale, _Scene, _Util } from 'ag-charts-community';
+import { PolarCrossLine } from '../polarCrossLine';
+
+const { ChartAxisDirection } = _ModuleSupport;
+const { Path, Sector, Text } = _Scene;
+const { normalizeAngle360, isNumberEqual } = _Util;
+export class AngleCrossLine extends PolarCrossLine {
```
Comment (alantreadway) [inline packages/ag-charts-enterprise/src/axes/angle-number/angleCrossLine.ts]: This seems to be an identical implementation to the `angle-category` axis implementation? Can we factor out the common code please?

File: packages/ag-charts-enterprise/src/series/radial-bar/radialBarSeries.ts [hunk 5]
```diff
@@ -0,0 +1,708 @@
...
+    maybeRefreshNodeData() {
+        const circleChanged = this.didCircleChange();
+        if (!circleChanged && !this.nodeDataRefresh) return;
+        const [{ nodeData = [] } = {}] = this.createNodeDataSync();
+        this.nodeData = nodeData;
+        this.nodeDataRefresh = false;
+    }
+
+    async createNodeData() {
+        return this.createNodeDataSync();
```
Comment (alantreadway) [inline packages/ag-charts-enterprise/src/series/radial-bar/radialBarSeries.ts:265]: This seems completely unnecessary? Please maintain the async contract - all callers (specifically `maybeRefreshNodeData()`) are called by `async` functions so there should be no need for a synchronous version of this function.

**Review by alantreadway — CHANGES_REQUESTED** [review]

File: packages/ag-charts-enterprise/src/series/radial-bar/radialBarSeries.ts (hunk 5 above)
Comment (alexanderby) [reply packages/ag-charts-enterprise/src/series/radial-bar/radialBarSeries.ts:265]: I saw it was implemented in Pie series that way. I will try making it async.

Comment (alexanderby) [reply packages/ag-charts-enterprise/src/series/radial-bar/radialBarSeries.ts:265]: Unfortunately there is `computeLabelsBBox` method of `PolarSeries` that is `sync`. Maybe I could make a follow up PR to fix it for all the polar series?

Comment (alantreadway) [reply packages/ag-charts-enterprise/src/series/radial-bar/radialBarSeries.ts:265]: Ok sounds good

**Review by alantreadway — APPROVED** [review]
LGTM - thanks for the updates, looks much better!

**Pull request merged**